src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -30,448 +32,397 @@ namespace keyframe {
+double computeSharpness(const cv::Mat& grayscaleImage, const int window_size)
 {
-  if((_maxOutFrame != 0) &&
-     !_hasSharpnessSelection &&
-     !_hasSparseDistanceSelection)
-  {
-      ALICEVISION_LOG_ERROR("KeyframeSelector needs at least one selection method if output frame limited !");
-      throw std::invalid_argument("KeyframeSelector needs at least one selection method if output frame limited !");
-  }
-
-  // load vocabulary tree
-  _voctree.reset(new aliceVision::voctree::VocabularyTree<DescriptorFloat>(voctreeFilePath));
-
-  {
-      ALICEVISION_LOG_INFO("vocabulary tree loaded with :" << std::endl
-                       << "\t- " << _voctree->levels() << " levels" << std::endl
-                       << "\t- " << _voctree->splits() << " branching factor" << std::endl);
-  }
-
-  // check number of input media filePaths
-  if(mediaPaths.empty())
-  {
-    ALICEVISION_LOG_ERROR("Cannot create KeyframeSelector without a media file path !");
-    throw std::invalid_argument("Cannot create KeyframeSelector without a media file path !");
-  }
-
-  // resize mediasInfo container
-  _mediasInfo.resize(mediaPaths.size());
-
-  // create SIFT image describer
-  _imageDescriber.reset(new feature::ImageDescriber_SIFT());
-}
+    cv::Mat sum, sumsq, laplacian;
+    cv::Laplacian(grayscaleImage, laplacian, CV_64F);
+    cv::integral(laplacian, sum, sumsq);
 
-void KeyframeSelector::process()
-{
-  // create feeds and count minimum number of frames
-  std::size_t nbFrames = std::numeric_limits<std::size_t>::max();
-  for(std::size_t mediaIndex = 0; mediaIndex < _mediaPaths.size(); ++mediaIndex)
-  {
-    const auto& path = _mediaPaths.at(mediaIndex);
+    double totalCount = window_size * window_size;
+    double maxstd = 0.0;
+    for (int y = 0; y < sum.rows - window_size; y++)
+    {
+        for (int x = 0; x < sum.cols - window_size; x++)
+        {
+            double tl = sum.at<double>(y, x);
+            double tr = sum.at<double>(y, x + window_size);
+            double bl = sum.at<double>(y + window_size, x);
+            double br = sum.at<double>(y + window_size, x + window_size);
+            double s1 = br + tl - tr - bl;
 
-    // create a feed provider per mediaPaths
-    _feeds.emplace_back(new dataio::FeedProvider(path));
+            tl = sumsq.at<double>(y, x);
+            tr = sumsq.at<double>(y, x + window_size);
+            bl = sumsq.at<double>(y + window_size, x);
+            br = sumsq.at<double>(y + window_size, x + window_size);
+            double s2 = br + tl - tr - bl;
 
-    const auto& feed = *_feeds.back();
+            double std_2 = std::sqrt((s2 - (s1 * s1) / totalCount) / totalCount);
 
-    // check if feed is initialized
-    if(!feed.isInit())
-    {
-      ALICEVISION_LOG_ERROR("Cannot initialize the FeedProvider with " << path);
-      throw std::invalid_argument("Cannot while initialize the FeedProvider with " + path);
+            maxstd = std::max(maxstd, std_2);
```

Comment:
If I understand correctly, with the max here, one pixel will give the score for the whole image? If that's the case, this does not sound reliable.